driver/src/metrics/stablex_metrics.rs

```diff
@@ -0,0 +1,195 @@
+                self.tokens
+                    .with_label_values(label)
+                    .inc_by(tokens_from_orders(&orders));
+                self.users
+                    .with_label_values(label)
+                    .inc_by(users_from_orders(&orders));
+            }
+            Err(_) => self.failures.with_label_values(label).inc(),
+        }
+    }
+
+    pub fn auction_solution_computed(
+        &self,
+        batch: U256,
+        orders: &Vec<Order>,
+        res: &Result<Solution, PriceFindingError>,
+    ) {
+        let label = &["solution"];
+        self.processing_times
+            .with_label_values(label)
+            .observe(time_elapsed_since_batch_start(batch));
+        match res {
+            Ok(solution) => {
+                let touched_orders = orders
+                    .iter()
+                    .zip(&solution.executed_buy_amounts)
+                    .filter(|(_, amount)| amount > &&0)
+                    .map(|(o, _)| o.clone())
+                    .collect::<Vec<Order>>();
+                self.orders
+                    .with_label_values(label)
+                    .inc_by(touched_orders.len().try_into().unwrap_or(std::i64::MAX));
+                self.tokens
+                    .with_label_values(label)
+                    .inc_by(tokens_from_orders(&touched_orders));
+                self.users
+                    .with_label_values(label)
+                    .inc_by(users_from_orders(&touched_orders));
+            }
+            Err(_) => self.failures.with_label_values(label).inc(),
+        }
+    }
+
+    pub fn auction_solution_verified(&self, batch: U256, res: &Result<U256, DriverError>) {
+        let label = &["verification"];
+        self.processing_times
+            .with_label_values(label)
+            .observe(time_elapsed_since_batch_start(batch));
+        match res {
+            Ok(_) => (),
+            Err(_) => self.failures.with_label_values(label).inc(),
+        }
+    }
+
+    pub fn auction_solution_submitted(&self, batch: U256, res: &Result<(), DriverError>) {
+        let label = &["submission"];
+        self.processing_times
+            .with_label_values(label)
+            .observe(time_elapsed_since_batch_start(batch));
+        match res {
+            Ok(_) => self.successes.with_label_values(label).inc(),
+            Err(_) => self.failures.with_label_values(label).inc(),
+        }
+    }
+
+    pub fn auction_skipped(&self, batch: U256) {
+        let label = &["skipped"];
+        self.processing_times
+            .with_label_values(label)
+            .observe(time_elapsed_since_batch_start(batch));
+        self.successes.with_label_values(label).inc();
+    }
+}
+
+fn time_elapsed_since_batch_start(batch: U256) -> f64 {
+    let now = Utc::now().timestamp() as u64;
+    // A new batch is created every 5 minutes and becomes solvable one batch later
+    let batch_start = (batch.low_u64() + 1) * 300;
```

Comment:
Is a batch a U256 that represents the unix time stamp in seconds or do the first 192 bits carry any meaning (are sometimes not 0)?

If its the first case then we could U256::as_u64 which panics if the value is too large.

Reply:
A batchID is defined as unixtimestamp_sec // 300 (5minutes), we really only need a few significant bits. In the [new contract](https://github.com/gnosis/dex-contracts/blob/2dfb10ea610e12109460c499c548361c6beb7ce1/contracts/EpochTokenLocker.sol#L145-L146) the data type is actually a u32